internal/ignition/ignition.go

```diff
@@ -411,3 +442,49 @@ func writeIgnitionFile(path string, config *config_31_types.Config) error {
 
 	return nil
 }
+
+func setFileInIgnition(config *config_31_types.Config, filePath string, fileContents string, mode int) error {
+	rootUser := "root"
+	file := config_31_types.File{
+		Node: config_31_types.Node{
+			Path:      filePath,
+			Overwrite: nil,
+			Group:     config_31_types.NodeGroup{},
+			User:      config_31_types.NodeUser{Name: &rootUser},
+		},
+		FileEmbedded1: config_31_types.FileEmbedded1{
+			Append: []config_31_types.Resource{},
+			Contents: config_31_types.Resource{
+				Source: &fileContents,
+			},
+			Mode: &mode,
+		},
+	}
+	config.Storage.Files = append(config.Storage.Files, file)
+	return nil
+}
+
+func setCACertInIgnition(role models.HostRole, path string, workDir string, caCertFile string) error {
+	config, err := parseIgnitionFile(path)
+	if err != nil {
+		return err
+	}
+
+	var caCertData []byte
+	caCertData, err = ioutil.ReadFile(caCertFile)
+	if err != nil {
+		return err
+	}
+
+	err = setFileInIgnition(config, common.ServiceCACert, fmt.Sprintf("data:,%s", url.PathEscape(string(caCertData))), 420)
+	if err != nil {
+		return err
+	}
+
+	fileName := fmt.Sprintf("%s.ign", role)
+	err = writeIgnitionFile(filepath.Join(workDir, fileName), config)
+	if err != nil {
+		return errors.Wrapf(err, "failed to write ignition for role %s", role)
```

Comment:
I don't think you need to wrap here as you have essentially the same error in the caller.